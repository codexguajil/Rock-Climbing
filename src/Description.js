import React, { Component } from 'react';

export default class Description extends Component {
  constructor() {
    super();
  }

  //toggle display description here

  render() {
    return (
      <div>
        <p>Hello</p>
      </div>
    )
  }
}