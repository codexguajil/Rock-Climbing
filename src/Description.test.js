import React from "react";
import Description from "./Description"
import { shallow } from "enzyme";

const mockObjectData = {
    "name": "Undertow",
    "id": 2,
    "grade": "5.10a",
    "type": ["trad", "aid", "alpine"],
    "link": "https://www.mountainproject.com/route/106200712/undertow"
}


describe("Description", () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
        <Description matchedType={mockObjectData} 
        />
    );
  });

  it("shold have a default state of false for toggle", () => {
      expect(wrapper.state()).toEqual({ showDescription: false})
  });

  it("should have a non expanded snapshot", () => {
      expect(wrapper).toMatchSnapshot()
  })

  it("should invoke the toggle function when list element is click", () => {
    expect(wrapper.state()).toEqual({ showDescription: false})
    wrapper.find('li').simulate('click')
    expect(wrapper.state()).toEqual({ showDescription: true})
  })

  it("should have a expanded sanpshot when funciton is invoke", () => {
    wrapper.find('li').simulate('click')
    expect(wrapper).toMatchSnapshot()
  })

});