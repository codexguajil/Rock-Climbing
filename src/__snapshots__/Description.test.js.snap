// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should have a expanded sanpshot when funciton is invoke 1`] = `
<li
  class="description-box"
  onClick={[Function]}
>
  <h3
    class="info"
  >
    Area: Estes Park Valley 
  </h3>
  <h4
    class="info"
  >
    Route Name: 
    Undertow
  </h4>
  <p
    class=" info toggle"
  >
     - 
  </p>
  <div
    class="info-box"
  >
    <p>
      Route Grade: 
      5.10a
    </p>
    <p>
      Route Type: 
      trad
      aid
      alpine
    </p>
    <a
      class="link"
      href="{this.props.matchedType.link}"
    >
      click here for more information on 
      Undertow
       
    </a>
  </div>
</li>
`;

exports[`Description should have a non expanded snapshot 1`] = `
<li
  class="description-box"
  onClick={[Function]}
>
  <h3
    class="info"
  >
    Area: Estes Park Valley 
  </h3>
  <h4
    class="info"
  >
    Route Name: 
    Undertow
  </h4>
  <p
    class=" info toggle"
  >
     + 
  </p>
</li>
`;
